include/robot_localization/ros_filter.hpp

```diff
@@ -199,14 +199,14 @@ class RosFilter : public rclcpp::Node
   //! @param[out] message - The standard ROS odometry message to be filled
   //! @return true if the filter is initialized, false otherwise
   //!
-  bool getFilteredOdometryMessage(nav_msgs::msg::Odometry * message);
+  bool getFilteredOdometryMessage(nav_msgs::msg::Odometry & message);
```

Comment:
I am struggling to find it, but at some point, there was definitely a document that suggested that in ROS 2, passing parameters as pointers was preferred. Was there a reason for this change? I'm not against it - I prefer reference parameters - but I am trying to determine if this was required for some reason, or you were just looking to be consistent.

Reply:
Raw pointers are more dangerous than references - pointer could be a NULL or contains wrong address, so it requires more guards in the method (e.g. https://pvs-studio.com/en/docs/warnings/v595/). So only reason is reference is safer because a compiler guards the argument.

> I am struggling to find it, but at some point, there was definitely a document that suggested that in ROS 

Unfortunately I didn't find that - but I found:

* https://design.ros2.org/articles/intraprocess_communications.html#number-of-message-copies -  based on what message should be used in intraprocess communication, subscriber with shared pointer generates less copies of message instead of  subscriber with unique pointer.
* https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/any_subscription_callback.hpp#L935 only callbacks with shared pointer generate less copies.

These links are not related with this comment but they could help with subscriptions in the future :)

Anyway I see, the best way is use [`std::optional`](https://en.cppreference.com/w/cpp/utility/optional) instead of reference and returning bool. Could you review my next commit?